python/ray/serve/tests/test_standalone2.py

```diff
@@ -1204,6 +1205,56 @@ async def waiter(*args):
         serve.shutdown()
 
 
+@pytest.mark.parametrize(
+    "ray_instance",
+    [
+        {
+            "LISTEN_FOR_CHANGE_REQUEST_TIMEOUT_S_LOWER_BOUND": "1",
+            "LISTEN_FOR_CHANGE_REQUEST_TIMEOUT_S_UPPER_BOUND": "2",
+        },
+    ],
+    indirect=True,
+)
+def test_long_poll_timeout_with_max_concurrent_queries(ray_instance):
+    """Test max_concurrent_queries can be honorded with long poll timeout
+
+    issue: https://github.com/ray-project/ray/issues/32652
+    """
+
+    signal_actor = SignalActor.remote()
+
+    @serve.deployment(max_concurrent_queries=1)
+    async def f():
+        await signal_actor.wait.remote()
+        return "hello"
+
+    handle = serve.run(f.bind())
+    first_ref = handle.remote()
+
+    # Second request should be hanging.
+    with ThreadPoolExecutor() as pool:
+        # Send the first request, it should block for the result
+        second_request_fut = pool.submit(
+            functools.partial(requests.get, "http://127.0.0.1:8000", timeout=100)
+        )
+        # Let the long poll timeout happens.
+        time.sleep(20)
```

Comment:
we should not be adding long sleeps to our test cases. is there a way we can test this deterministically instead?

if not, at a bare minimum we should make this timeout much tighter so as not to dramatically slow down our tests

Reply:
I would still prefer to not have a timing dependency here.

Is it possible to have a manual call to flush the `in_flight_queries`, going through the same path as the timeout?

I'm fine with doing this in a separate PR, it might require some refactoring... but it's a red flag if we need to rely on sleeps for testing